sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1230,11 +1236,13 @@ def create(
         **render_kwargs: t.Any,
     ) -> None:
         model = snapshot.model
-        assert isinstance(model.kind, SCDType2Kind)
+        assert isinstance(model.kind, (SCDType2ByTimeKind, SCDType2ByColumnKind))
         if model.annotated:
             logger.info("Creating table '%s'", name)
             columns_to_types = model.columns_to_types_or_raise
-            columns_to_types[model.kind.updated_at_name] = model.kind.time_data_type
+            if model.kind.is_scd_type_2_by_time:
+                assert isinstance(model.kind, SCDType2ByTimeKind)
```

Comment:
why do you need these asserts? for mypy updated_at_name?

Reply:
given how many times you have to do assert isinstance to please mypy i'm inclined to believe there's probably a better way to accomplish this. i think perhaps the problem is the reliance on these helper methods is_scd, 

if instead you just did

if isinstance(model.kind, ScdType2ByTimeKind):
   
this would be cleaner, thoughts @izeigerman ?

Reply:
Yes for mypy

Reply:
Updated: https://github.com/TobikoData/sqlmesh/pull/1997/commits/97aa0d5ab6d30dbd9ffdae5c89a8ccae7e64f0dd